src/main/scala/interfaces/Converter.scala

```diff
@@ -0,0 +1,318 @@
+package viper.silicon.interfaces
+
+import viper.silver.verifier.{Model, ModelEntry, SingleEntry, MapEntry, ModelParser}
+import viper.silver.ast
+
+import viper.silicon.interfaces.state.Chunk
+import viper.silicon.resources.{FieldID, PredicateID}
+import viper.silicon.state.{Store, State, BasicChunk}
+import viper.silicon.state.terms.{sorts, Sort, Term, Unit, IntLiteral, BooleanLiteral, Null, Var, App, Combine,
+                                  First, Second, SortWrapper, PredicateLookup, toSnapTree, Rational, PermLiteral}
+
+
+/* Some new classes to describe a more informative model */
+
+/* Explanation: (to be removed later)
+    To use these new extracted counterexamples, you can use the flag "--counterexample mapped"
+    The new trait ExtractedModelEntry should make the counterexamples more readable and extract
+    information from the heap, but heap values such as predicates can still be accessed,
+    also in a more processed way since all the first/second/Sortwrappers should be gone now and 
+    replaced with their corresponding values in the extracted Heap fields of the converter class
+
+    A lot of types like sequences etc are not handled yet, and I am not sure if I plan to do so
+    since they are not used in Rust
```

Comment:
It might be difficult for you to test, but we do have plans for more Viper types exposed to Prusti, see https://github.com/viperproject/prusti-dev/pull/181